forecasting/src/autogluon/forecasting/trainer/abstract_trainer.py

```diff
@@ -0,0 +1,320 @@
+import time
+import copy
+from ..models.gluonts_model.gluonts_abstract.gluonts_abstract_model import GluonTSAbstractModel
+import networkx as nx
+import logging
+from core.utils.savers import save_pkl, save_json
+from core.utils.loaders import load_pkl
+from collections import defaultdict
+import pandas as pd
+from gluonts.evaluation import Evaluator
+from gluonts.evaluation.backtest import make_evaluation_predictions
+
+
+logger = logging.getLogger(__name__)
+
+__all__ = ['AbstractTrainer']
+
+
+class AbstractTrainer:
+
+    trainer_file_name = 'trainer.pkl'
+    trainer_info_name = 'info.pkl'
+    trainer_info_json_name = 'info.json'
+
+    def __init__(self, path: str, freq, prediction_length, scheduler_options=None, eval_metric=None,
+                 save_data=False):
+        self.path = path
+        self.freq = freq
+        self.prediction_length = prediction_length
+        self.save_data = save_data
+
+        # do we really need the graph here as what is done in tabular?
+        self.model_graph = nx.DiGraph()
```

Comment:
probably dont need this?

Reply:
yes, so I deleted it on my laptop and will push it soon.